app/src/main/java/app/gamenative/ui/screen/settings/SettingsGroupInterface.kt

```
@@ -353,20 +354,34 @@ fun SettingsGroupInterface(
-                .filter { Environment.getExternalStorageState(it) == Environment.MEDIA_MOUNTED }
-                .filter { sm.getStorageVolume(it)?.isPrimary != true }
+            // ContextCompat.getExternalFilesDirs returns ALL app-specific dirs.
+            // It returns NULL for volumes that are not mounted, so we just filterNotNull().
+            val allVolumes = ContextCompat.getExternalFilesDirs(ctx, null).filterNotNull()
+
+            // Filter 1: Identify SD Cards/USB by checking if they are "Removable"
+            // This replaces the complex StorageManager/isPrimary logic
+            allVolumes.filter { dir ->
+                try {
+                    Environment.isExternalStorageRemovable(dir)
+                } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)